ParticleObjects/Source/surfaceSource_class.f90

```diff
@@ -0,0 +1,324 @@
+module surfaceSource_class
```

Comment:
I might rename this to be less general sounding. Add a blackBody in there for example.